source/components/Avatar/index.js

```diff
@@ -7,20 +7,31 @@ import { getAvatarImage } from './utils';
 
 import './styles.scss';
 
-const Avatar = ({
-    alt,
-    badge,
-    className,
-    href,
-    src,
-    title,
-    linkBuilder,
-}) => (
-    <div className={classNames('wds-avatar', className)} title={title}>
-        {getAvatarImage(href, alt, src, linkBuilder)}
-        {badge && <Badge badge={badge} />}
-    </div>
-);
+class Avatar extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            imageSrc: props.src,
+        };
+    }
+
+    componentDidMount() {
+        if (!this.state.imageSrc) {
+            fetch(`https://services.wikia.com/user-attribute/user/${this.props.userId}/attr/avatar`)
```

Comment:
I was imagining using React.Context to hold all the avatar URLs so apps only fetch once during the lifetime of the app. I think that is a secondary feature but if you believe it will be straight forward to include. That would be a neat improvement.